event/consumer.go

```diff
@@ -2,38 +2,38 @@ package event
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/ONSdigital/dp-import-cantabular-dataset/config"
 	"github.com/ONSdigital/dp-import-cantabular-dataset/schema"
+
 	kafka "github.com/ONSdigital/dp-kafka/v2"
-	"github.com/ONSdigital/log.go/log"
+	"github.com/ONSdigital/log.go/v2/log"
 )
 
-//go:generate moq -out mock/handler.go -pkg mock . Handler
-
-// TODO: remove or replace hello called logic with app specific
-// Handler represents a handler for processing a single event.
-type Handler interface {
-	Handle(ctx context.Context, cfg *config.Config, helloCalled *HelloCalled) error
-}
-
 // Consume converts messages to event instances, and pass the event to the provided handler.
-func Consume(ctx context.Context, messageConsumer kafka.IConsumerGroup, handler Handler, cfg *config.Config) {
-
+func Consume(ctx context.Context, cg kafka.IConsumerGroup, h Handler, cfg *config.Config) {
 	// consume loop, to be executed by each worker
 	var consume = func(workerID int) {
 		for {
 			select {
-			case message, ok := <-messageConsumer.Channels().Upstream:
+			case msg, ok := <-cg.Channels().Upstream:
 				if !ok {
-					log.Event(ctx, "closing event consumer loop because upstream channel is closed", log.INFO, log.Data{"worker_id": workerID})
+					log.Info(ctx, "upstream channel closed - closing event consumer loop", log.Data{"worker_id": workerID})
 					return
 				}
-				messageCtx := context.Background()
-				processMessage(messageCtx, message, handler, cfg)
-				message.Release()
-			case <-messageConsumer.Channels().Closer:
-				log.Event(ctx, "closing event consumer loop because closer channel is closed", log.INFO, log.Data{"worker_id": workerID})
+
+				if err := processMessage(context.Background(), msg, h, cfg); err != nil{
+					log.Error(ctx, "failed to process message", err, log.Data{
+						"status_code": statusCode(err),
+						"log_data": unwrapLogData(err),
+					})
+					// Need to send response to import-api to notify of failure.
+				}
+
+				msg.Release()
```

Comment:
I see that you removed the Commit logic from `processMessage`, in this case, we should call `msg.CommitAndRelease()` here instead of only Release.